gpytorch/kernels/cosine_kernel.py

```diff
@@ -57,19 +57,18 @@ class CosineKernel(Kernel):
     def __init__(
         self, active_dims=None, batch_size=1, log_period_length_prior=None, eps=1e-6, param_transform=torch.exp
     ):
-        super(CosineKernel, self).__init__(
-            has_lengthscale=False, active_dims=active_dims, param_transform=param_transform
-        )
+        super(CosineKernel, self).__init__(has_lengthscale=False, active_dims=active_dims)
         self.eps = eps
         self.register_parameter(
             name="log_period_length",
             parameter=torch.nn.Parameter(torch.zeros(batch_size, 1, 1)),
             prior=log_period_length_prior,
+            transform=param_transform,
```

Comment:
I guess I'm not a huge fan of putting the transform on the `register_parameter` call, because now our GPyTorch params are getting further removed from PyTorch params.

It's also a little weird with the naming. For example, `self.log_period_length` returns the log of the period length, even though it knows about the parameter transform.

Reply:
I can see why we wouldn't want to make parameters different from torch ones